llvm/lib/Analysis/InstructionSimplify.cpp

```diff
@@ -48,7 +48,7 @@ using namespace llvm::PatternMatch;
 
 #define DEBUG_TYPE "instsimplify"
 
-enum { RecursionLimit = 3 };
+static unsigned RecursionLimit = 5;
```

Comment:
```suggestion
static constexpr unsigned RecursionLimit = 5;
```

@nikic Can you test the compile-time impact?

TBH I don't like increasing the recursion limit for all root instructions.
If necessary, it would be better to only increase the limit for `shufflevector`:
```
/// Given operands for a ShuffleVectorInst, fold the result or return null.
Value *llvm::simplifyShuffleVectorInst(Value *Op0, Value *Op1,
                                       ArrayRef<int> Mask, Type *RetTy,
                                       const SimplifyQuery &Q) {
  static constexpr ShuffleRecursionLimit = 5;
  return ::simplifyShuffleVectorInst(Op0, Op1, Mask, RetTy, Q, ShuffleRecursionLimit);
}
```

Reply:
I'm surprised there's a threshold here that isn't 6

Reply:
Compile-time: http://llvm-compile-time-tracker.com/compare.php?from=f6527774569790b5a5236f6e84f3f839ce6c2fff&to=4363c34b550a0908edf4ee89eda2fee72a096ba9&stat=instructions:u